wgpu-core/src/command/compute.rs

```diff
@@ -6,237 +6,294 @@ use crate::{
     command::{
         bind::{Binder, LayoutChange},
         CommandBuffer,
+        PhantomSlice,
     },
     device::{all_buffer_stages, BIND_BUFFER_ALIGNMENT},
-    hub::{GfxBackend, Global, IdentityFilter, Token},
-    id::{BindGroupId, BufferId, CommandBufferId, ComputePassId, ComputePipelineId},
+    hub::{GfxBackend, Global, Token},
+    id,
     resource::BufferUsage,
-    track::TrackerSet,
     BufferAddress,
-    Stored,
 };
 
-use hal::{self, command::CommandBuffer as _};
+use hal::command::CommandBuffer as _;
+use peek_poke::{Peek, PeekCopy, Poke};
 
 use std::iter;
 
-#[derive(Debug)]
-pub struct ComputePass<B: hal::Backend> {
-    raw: B::CommandBuffer,
-    cmb_id: Stored<CommandBufferId>,
-    binder: Binder,
-    trackers: TrackerSet,
-}
 
-impl<B: hal::Backend> ComputePass<B> {
-    pub(crate) fn new(
-        raw: B::CommandBuffer,
-        cmb_id: Stored<CommandBufferId>,
-        trackers: TrackerSet,
-        max_bind_groups: u32,
-    ) -> Self {
-        ComputePass {
-            raw,
-            cmb_id,
-            binder: Binder::new(max_bind_groups),
-            trackers,
-        }
-    }
+#[derive(Clone, Copy, Debug, PeekCopy, Poke)]
```

Comment:
These don't need to be repr(C) because we never lay them out in memory that C code works with.

Reply:
I thought peek_poke expects to operate on `repr(C)` types to have certain guarantees about the data layout, or at least the peek_poke examples seem to do that

Reply:
hmm, I'll double check!

Reply:
The webrender examples uses repr(C) because it was written for Gecko C++ code to see the types and copied verbatim into peek-poke.

That is to say, I consulted with @Gankra and it turns out there can be a slightly bigger chance of the compiler optimization (turning the struct poking into a memcopy) if it's `repr(C)`. Still not a guarantee, still not a correctness concern, just a bit more likely.

So we should make these types `repr(C)` as you suggest, at least the `Rect` thing.